### PR TITLE
networking/validation: add endport range validation

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add endPort range validation per request in the initial PR https://github.com/kubernetes/kubernetes/pull/97058/files#r568275888

/assign @rikatz @jayunit100 @abhiraut

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/networking/validation/validation.go [hunk 1]
```diff
@@ -68,8 +68,13 @@ func ValidateNetworkPolicyPort(port *networking.NetworkPolicyPort, portPath *fie
 			for _, msg := range validation.IsValidPortNum(int(port.Port.IntVal)) {
 				allErrs = append(allErrs, field.Invalid(portPath.Child("port"), port.Port.IntVal, msg))
 			}
-			if port.EndPort != nil && *port.EndPort < port.Port.IntVal {
-				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "must be greater than or equal to `port`"))
+			if port.EndPort != nil {
+				if *port.EndPort < port.Port.IntVal {
+					allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "must be greater than or equal to `port`, not higher than 65535"))
```
Comment (abhiraut) [inline pkg/apis/networking/validation/validation.go:73]: nit: if violated, the 65535 range inclusive error will be appended by isValidPortNum()..

Comment (cpanato) [reply pkg/apis/networking/validation/validation.go]: done, thanks for the review

**Comment by abhiraut** [comment]
looks good to me

File: pkg/apis/networking/validation/validation_test.go [hunk 2]
```diff
@@ -1018,6 +1018,32 @@ func TestValidateNetworkPolicy(t *testing.T) {
 				},
 			},
 		},
+		"invalid endport range defined": {
```
Comment (thockin) [inline pkg/apis/networking/validation/validation_test.go:1021]: Not going to hold this PR to it, but this is a great example of a place where some helper functions could go a LONG way.

E.g. this should be something like:

```
"invalid endport range defined": makeService(egress(port(intstr.FromInt(30000), 65537)))
```

Everything else seems like boilerplate.  Repeat that for every test case in this one test and delete hundreds of LOC

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

File: pkg/apis/networking/validation/validation_test.go (hunk 2 above)
Comment (cpanato) [reply pkg/apis/networking/validation/validation_test.go:1021]: will do in a follow-up, thanks Tim for the feedback

**Pull request merged**